producer/src/cache.rs

```diff
@@ -0,0 +1,219 @@
+use identity_iota::did::CoreDID;
+use identity_iota::document::CoreDocument;
+use log::info;
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+    time::{Duration, Instant},
+};
+
+/// Caches the resolved DID Documents in-memory.
+///
+/// * **simple**: key-value store where the key is the `DID` and the value is the resolved `DID Document`
+/// * **lazy**: the cache does not automatically run any clean up tasks. It evaluates the cache entries only when they are accessed.
+///
+/// NOTE: Use with caution! Caches can hold outdated data. A short TTL is recommended.
+#[derive(Clone, Debug)]
+pub struct InMemoryCache {
+    entries: Arc<Mutex<HashMap<CoreDID, CacheEntry>>>,
```

Comment:
Not a fan of using `Arc<Mutex<T>>` here if it is not really crucial for `InMemoryCache`'s core functionality (which I think it isn't). 

If this was actually a workaround to satisfy the `Clone` derive, then I suggest to just remove that constraint since I don't think it needs to be cloned. You can remove it from `SecretManager` as well since I don't think any of our downstream repo's require `SecretManager` to be cloneable. If they do require it then `SecretManager` should be wrapped into an `Arc<Mutex<SecretManager>>` over there.